codegen/src/main/java/software/amazon/awssdk/codegen/internal/Utils.java

```diff
@@ -307,6 +309,10 @@ public static ShapeModel findShapeModelByC2jNameIfExists(IntermediateModel inter
         return null;
     }
 
+    public static List<ShapeModel> findShapesByC2jName(IntermediateModel intermediateModel, String shapeC2jName) {
+        return intermediateModel.getShapes().values().stream().filter(s -> s.getC2jName().equals(shapeC2jName)).collect(toList());
```

Comment:
In what cases would there be multiple?

Reply:
How do you know which is the right one from the list returned?

Reply:
We apply the mutation to all of the ShapeModels that are returned; in theory it's more deterministic than it was before where it just applied it to the "first" one (whichever that happened to be).

Reply:
Okay so the mutation for the original shape is just unused right?

Reply:
Yup, but at this point we don't know which one will be the unused one.

Reply:
Got it. Can we add a TODO somewhere to remove shapes when we rename them, that's hella confusing and i'm sure will cause future bugs.

Reply:
That might be easier said than done. Some shapes are legitimately re-used for more than one thing. Even our own tests do this where we have two operations that share a C2J shape:
https://github.com/aws/aws-sdk-java-v2/blob/master/codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/c2j/json/service-2.json#L67-L89

This will create two ShapeModels `StreamingInputOperationRequest` and `StreamingOutputOperationResponse` that both have the same C2jShape name (`StructureWithStreamingMember`)

Reply:
Good point, we should probably be more consistent about what identifiers we use for customizations. For things that are mutating the C2J representation we use the C2j name and that happens before the intermediate model is created and new shapes are added, and for everything else we use the Java class name which is guaranteed to be unique.